core/templates/domain/recommendations/exploration-recommendations-backend-api.service.ts

```diff
@@ -52,8 +52,11 @@ export class ExplorationRecommendationsBackendApiService {
   async getRecommendedSummaryDictsAsync(
       authorRecommendedExpIds: string[],
       includeSystemRecommendations: string,
-      collectionId: string, storyId: string, currentNodeId: string,
-      explorationId: string): Promise<LearnerExplorationSummary[]> {
+      // 'collectionId', 'storyId' and 'currentNodeId' are optional
+      // since they may not be present in the URL.
```

Comment:
Are they null or optional ?
if null:
```suggestion
      // 'collectionId', 'storyId' and 'currentNodeId' can be null
      // since they may not be present in the URL.
```
If they were optional the type would've been like this `collectionId?: string`

Reply:
done